pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,232 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+
+	"github.com/openshift/osde2e/pkg/common/aws"
+	"github.com/openshift/osde2e/pkg/common/expect"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/label"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+)
+
+// NodePool STS Permissions Test
+// This test validates day 2 operations for ROSA HCP clusters by creating NodePools.
+// It ensures required STS permissions (ec2:RunInstances, ec2:CreateTags, ec2:DescribeInstances) 
+
+var nodePoolTestName = "[Suite: e2e] NodePool STS Permissions"
+
+var _ = ginkgo.Describe(nodePoolTestName, ginkgo.Ordered, label.E2E, func() {
+	var h *helper.H
+	var client *resources.Resources
+
+	ginkgo.BeforeAll(func() {
+		h = helper.New()
+		client = h.AsUser("")
+	})
+
+	ginkgo.It("should create nodepool and provision EC2 instances with proper STS permissions", func(ctx context.Context) {
+		var nodes corev1.NodeList
+		expect.NoError(client.List(ctx, &nodes))
+		
+		if len(nodes.Items) == 0 {
+			ginkgo.Skip("No nodes found")
+		}
+
+		clusterName := extractClusterName(nodes.Items)
+		clusterID := getClusterID(nodes.Items)
+		
+		if clusterName == "" || clusterID == "" {
+			ginkgo.Skip("Could not determine cluster information")
```

Comment:
I think it would be helpful if this string was more specific to what is missing.